File: LEETCODE/LEETCODE-CPP/1726.cpp
//1726. Tuple with Same Product
//https://leetcode.com/problems/tuple-with-same-product/

class Solution {
public:
    int tupleSameProduct(vector<int>& nums) {
        int n=nums.size(),ans=0;
        unordered_map<int,int>mp;

        for(int i=0;i<n-1;i++){
            for(int j=i+1;j<n;j++){
                int temp= nums[i]*nums[j];

                ans+=8*mp[temp];
                mp[temp]++;
            }
        }

        return ans;
    }
};

File: LEETCODE/LEETCODE-PYTHON/1726.py
#1726. Tuple with Same Product
#https://leetcode.com/problems/tuple-with-same-product/description/

class Solution(object):
    def tupleSameProduct(self, nums):
        n=len(nums)
        ans=0

        dict=defaultdict(int)

        for i in range(0,n-1):
            for j in range(i+1,n):
                temp= nums[i]*nums[j]

                ans+=8*dict[temp]

                dict[temp]+=1

        return ans
        